src/meta/processors/job/JobManager.cpp

```diff
@@ -409,32 +409,60 @@ nebula::cpp2::ErrorCode JobManager::addJob(const JobDescription& jobDesc, AdminC
 }
 
 size_t JobManager::jobSize() const {
-  return highPriorityQueue_->size() + lowPriorityQueue_->size();
+  size_t size = 0;
+  for (auto iter = priorityQueues_.begin(); iter != priorityQueues_.end(); ++iter) {
+    size += iter->second->size();
+  }
+  return size;
 }
 
-bool JobManager::try_dequeue(std::pair<JbOp, JobID>& opJobId) {
-  if (highPriorityQueue_->try_dequeue(opJobId)) {
-    return true;
-  } else if (lowPriorityQueue_->try_dequeue(opJobId)) {
-    return true;
+// Execute jobs concurrently between spaces
+// Execute jobs according to priority within the space
+bool JobManager::tryDequeue(std::tuple<JbOp, JobID, GraphSpaceID>& opJobId) {
+  for (auto queueIter = priorityQueues_.begin(); queueIter != priorityQueues_.end(); ++queueIter) {
+    auto spaceId = queueIter->first;
+    auto runningJobIter = spaceRunningJobs_.find(spaceId);
+    if (runningJobIter != spaceRunningJobs_.end() && runningJobIter->second) {
+      continue;
+    }
+    if (queueIter->second == nullptr) {
+      spaceRunningJobs_.insert_or_assign(spaceId, false);
+      continue;
+    }
+    auto ret = queueIter->second->try_dequeue(opJobId);
+    if (ret) {
+      return true;
+    }
   }
   return false;
 }
 
-void JobManager::enqueue(const JbOp& op, const JobID& jobId, const cpp2::JobType& jobType) {
-  if (jobType == cpp2::JobType::STATS) {
-    highPriorityQueue_->enqueue(std::make_pair(op, jobId));
+void JobManager::enqueue(GraphSpaceID space,
+                         JobID jobId,
+                         const JbOp& op,
+                         const cpp2::JobType& jobType) {
+  auto iter = priorityQueues_.find(space);
+  if (iter == priorityQueues_.end()) {
+    std::unique_ptr<PriorityQueue> priQueue = std::make_unique<PriorityQueue>(2);
```

Comment:
Better write an enum class to expresss the magic num (0, 1, 2)

Reply:
ok